### ISSUE TITLE
Config_get tries to obtain value from config before config has been set

### ISSUE THREAD

**Issue by knservis** [rucio/rucio#3910]
Trying to run `bin/rucio --help` in order to find out the arguments needed for config, fails with:
```
Traceback (most recent call last):
  File "bin/rucio", line 2352, in <module>
    oparser = get_parser()
  File "bin/rucio", line 2151, in get_parser
    upload_parser.add_argument('--transfer-timeout', dest='transfer_timeout', type=float, action='store', default=config_get('upload', 'transfer_timeout', False, 3600), help='Transfer timeout (in seconds).')
  File "/rucio/lib/rucio/common/config.py", line 52, in config_get
    return get_config().get(section, option)
  File "/rucio/lib/rucio/common/config.py", line 211, in get_config
    __CONFIG = Config()
  File "/rucio/lib/rucio/common/config.py", line 237, in __init__
    '\n\t' + '\n\t'.join(configs))
RuntimeError: Could not load Rucio configuration file. Rucio looked in the following paths for a configuration file, in order:
        /opt/rucio/etc/rucio.cfg
```

The reason is this line that tries to get a default from the config, before the config has been set:

https://github.com/rucio/rucio/blob/7820df402a54bcf1436173fa517f79b9bcc8986b/bin/rucio#L2151

**Comment by knservis** [comment]
A similar thing happens when running `bin/rucio-admin --help`

```                                                                                                                                                                       
Traceback (most recent call last):                                                                                                                                                             
  File "bin/rucio-admin", line 70, in <module>                                                                                                                                                 
    from rucio.rse import rsemanager as rsemgr                                                                                                                                                 
  File "/rucio/lib/rucio/rse/__init__.py", line 32, in <module>                                                                                                  
    if config.config_has_section('database'):                                                                                                                                                  
  File "/rucio/lib/rucio/common/config.py", line 66, in config_has_section                                                                                       
    return get_config().has_section(section)                                                                                                                                                   
  File "/rucio/lib/rucio/common/config.py", line 211, in get_config                                                                                              
    __CONFIG = Config()                                                                                                                                                                        
  File "/rucio/lib/rucio/common/config.py", line 237, in __init__                                                                                                
    '\n\t' + '\n\t'.join(configs))                                                                                                                                                             
RuntimeError: Could not load Rucio configuration file. Rucio looked in the following paths for a configuration file, in order:                                                                 
        /opt/rucio/etc/rucio.cfg
```

Although in this case I don't see a `--config` option so it may be expected to fail if RUCIO_CONFIG env var is not set.

**Comment by TomasJavurek** [comment]
I do not consider it as a bug. The config wasn't set, so it correctly returned a propper exception. We could for example catch it in the get_config() factory method, and print additional info, e.g. Please, set the configuration file - Check /rucio/etc/rucio.cfg.template. Or, we can print out a minimum config file but that might be way too long.

**Comment by bziemons** [comment]
I think we should do everything so that the config is not used on mere using `--help` or importing modules, because in most cases it is not necessary to get a config value until you actually use a command that needs the config. It is a big plus from the usability point of view. Both when using the Rucio CLI or when using Rucio as a library.

**Comment by knservis** [comment]
Running a command with `--help` (one reason is to find out the argument to provide a config) normally returns a help text and does not throw an exception in every other program I have used. I would say that `--help` should be processed first and then at runtime (not loading) the config should be loaded and and validated in one location. That helps provide user feedback on what is wrong with the config. I would imagine that helps with testing too.

**Comment by TomasJavurek** [comment]
> I think we should do everything so that the config is not used on mere using `--help` or importing modules, because in most cases it is not necessary to get a config value until you actually use a command that needs the config. It is a big plus from the usability point of view. Both when using the Rucio CLI or when using Rucio as a library.

I see what can I do. There might be a complication. We sometimes need the config in constructors in order to distinguish the client/server mode.

**Comment by bari12** [comment]
This is actually somewhat of a duplicate of #1911 
I agree as well, a simple `--help` should not fail because the configuration is not there. The user might even want to get help on how to setup the configuration ;-)
That it is probably not that easy to change due to the code flow we currently have is true, but nevertheless it shouldn't be the case.

**Comment by bari12** [comment]
Fixed in #1911
